File: app/src/main/java/com/example/architechturestartercode/network/Network.java
package com.example.architechturestartercode.network;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class Network {
    private MoviesService moviesService;
    private static Network instance;

    private Network() {
    }

    public static Network getInstance() {
        if (instance == null) {
            instance = new Network();
        }
        return instance;
    }

    public MoviesService getMoviesService() {
       Retrofit retrofit = new Retrofit.Builder()
                .baseUrl("https://dummyjson.com/")
                .addConverterFactory(GsonConverterFactory.create())
                .build();
        return retrofit.create(MoviesService.class);
    }
}

File: app/src/main/java/com/example/architechturestartercode/network/Network.kt
package com.example.architechturestartercode.network

import com.example.architechturestartercode.data.movie.datasource.remote.MovieService
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class Network private constructor()  {
     var movieService: MovieService?

    init {
        val retrofit = Retrofit.Builder()
            .baseUrl("https://api.themoviedb.org/3/")
            .addConverterFactory(GsonConverterFactory.create())
            .build()

        movieService = retrofit.create<MovieService?>(MovieService::class.java)
    }

    companion object {
        var IMAGE_BASE_URL: String = "https://image.tmdb.org/t/p/w500"
        var instance: Network? = null
            get() {
                if (field == null) {
                    field = Network()
                }
                return field
            }
            private set
    }
}
